src/main/java/seedu/address/ui/ShiftCard.java

```diff
@@ -27,22 +29,27 @@
     @FXML
     private FlowPane roleRequirements;
     @FXML
-    private FlowPane workerRoleAssignments;
+    private FlowPane shiftAssignments;
 
     /**
      * Creates a {@code ShiftCard} with the given {@code Shift} and index to display.
      */
-    public ShiftCard(Shift shift, int displayedIndex) {
+    public ShiftCard(Shift shift, int displayedIndex, ObservableList<Assignment> assignmentList) {
         super(FXML);
         this.shift = shift;
         id.setText(displayedIndex + ". ");
         dayTime.setText(shift.getShiftDay().toString() + " " + shift.getShiftTime().toString());
         shift.getRoleRequirements().stream()
                 .sorted(Comparator.comparing(roleRequirement -> roleRequirement.getRole().getRole()))
                 .forEach(roleRequirement -> roleRequirements.getChildren().add(new Label(roleRequirement.toString())));
-        shift.getWorkerRoleAssignments()
-                .forEach(workerRoleAssignment -> workerRoleAssignments.getChildren().add(new Label(
-                        workerRoleAssignment.getWorker().getName() + " " + workerRoleAssignment.getRole())));
+
+        assignmentList.forEach(assignment -> {
```

Comment:
Maybe if these are created as proper JavaFX nodes with assignment set to its inner value, then JavaFX will update them properly. Implementation could be similar to `WorkerListPanel` except it is as children to `WorkerCard` and `ShiftCard`.

Reply:
Great idea, thank you!